### ISSUE TITLE
[bndtools] Resolve not working

### ISSUE THREAD

**Issue by kriegfrj** [bndtools/bnd#3791]
It seems that resolving in the GUI is not working in the current snapshot build of Bndtools:

* With resolution set to "manual", I click on the "Resolve" button, but nothing happens (not even a progress meter to indicate that resolution is in progress).
* With resolution set to "auto", I cannot save the `.bndrun` file. The save fails silently.

There are no clues in the Eclipse Error Log, nor in the Eclipse Host Console.

If no one else is seeing this, I'll be happy to investigate.

**Comment by bjhargrave** [comment]
I have made a number of changed to the cap/req generation code (e.g. ResourceBuilder) recently and caps/reqs are fed into the resolver. But I am not sure if that is involved here. If you can look into this, that would be great!

**Comment by kriegfrj** [comment]
Ok, I think that this may be related to #3779.

Using the debugger, I've tracked it down to this bit of code:

https://github.com/bndtools/bnd/blob/ef1943777a51f2d41ac4fc34119cc944aeac95fb/bndtools.core/src/bndtools/editor/BndEditor.java#L278-L282

I've identified two code paths:

* `BndEditor.doSave()` (runs in resolve: auto mode when you save the page)
* `bndtools.editor.project.RunRequirementsPart.doResolve()` (runs when you click the resolve button)

There are two issues (the same issues in both code paths):
1. `inputResource != null`, so the failure is returned and the resolution job is never started (and in the case of doSave(), the chained save job is also never started and so the save is effectively aborted).
2. The result code of `resolveRunBundles()` is silently ignored instead of being reported to the user.

Both of these things need to be fixed. The first could be as simple as changing != into ==, but I'll need to have a closer look at the code to confirm (hoping @pkriens might be able to short-circuit this for me, given that he was working on that code recently...).

Confirming again how nice it would be to have some regression tests for the GUI...
